pkg/volume/util/operationexecutor/operation_generator.go

```diff
@@ -723,12 +744,26 @@ func (og *operationGenerator) GenerateUnmountDeviceFunc(
 	deviceToDetach AttachedVolume,
 	actualStateOfWorld ActualStateOfWorldMounterUpdater,
 	mounter mount.Interface) (volumetypes.GeneratedOperations, error) {
+
+	var pluginName string
+	if useCSIPlugin(og.volumePluginMgr, deviceToDetach.VolumeSpec) {
+		pluginName = csi.CSIPluginName
+		csiSpec, err := translateSpec(deviceToDetach.VolumeSpec)
```

Comment:
do we need to do any of this translation work if we're not using it afterwards?

Reply:
csiSpec used after the error check:
```
deviceToDetach.VolumeSpec = csiSpec
```